index.js

```diff
@@ -0,0 +1,166 @@
+ * response doesn't look correct.
+ *
+ * Currently, only two parameters are supported with the ad placement: publisher
+ * id and the place type. All of this is determined by the server and this
+ * client so far only renders the API return HTML.
+ *
+ * This can be loaded async. CSS styles are preloaded via webpack
+ * `to-string-loader`.
+ *
+ * Usage:
+ *
+ *     <script async src="ethicalads.min.js"></script>
+ *     <div data-ethical-ad-publisher="foo" data-ethical-ad-type="text"></div>
+ */
+
+import * as ad_css from "./styles.scss";
+
+const AD_DECISION_URL = "https://server.ethicalads.io/api/v1/decision/";
+
+/* Placement object to query decision API and return an Element node
+ *
+ * @param publisher
+ * @param ad_type
+ */
+export class Placement {
+  constructor(publisher, ad_type = "readthedocs-sidebar") {
+    this.publisher = publisher;
+    this.ad_type = ad_type;
+  }
+
+  /* Load placement from decision API and create DOM element
+   *
+   * @returns Promise
+   */
+  load() {
+    const id = "ad_" + Date.now();
+    const url_params = new URLSearchParams({
+      publisher: this.publisher,
+      ad_types: this.ad_type,
+      div_ids: id,
+      callback: id,
+      format: "jsonp",
+    });
+    const url = new URL(AD_DECISION_URL + "?" + url_params.toString());
+    const promise = new Promise((resolve, reject) => {
+      window[id] = (response) => {
+        if (response && response.id) {
+          return resolve(response);
+        } else {
+          return reject(
+            new Error("Placement is configured with invalid parameters.")
+          );
+        }
+      };
+
+      var script = document.createElement("script");
+      script.src = url;
+      script.type = "text/javascript";
+      script.async = true;
+      script.addEventListener("error", reject);
+      document.getElementsByTagName("head")[0].appendChild(script);
+    }).then((response) => {
+      const element = document.createElement("div", { id: this.id });
+      element.innerHTML = response.html;
+
+      var impression_view = document.createElement("img", {
+        src: response.view_url,
+        style: "display: none;",
+      });
+      element.appendChild(impression_view);
+
+      return element;
+    });
+
+    return promise;
+  }
+}
+
+/* Find all placement DOM elements and hot load HTML as child nodes
+ *
+ * @returns Promise
+ */
+export function load_placements() {
+  // Find all elements matching required data binding attribute
+  const node_list = document.querySelectorAll("[data-ethical-ad-publisher]");
+  const elements = Array.prototype.slice.call(node_list);
+
+  // Create main promise. Iterator `all()` Promise wil surround array of found
+  // elements. If any of these elements have issues, this main promise will
+  // reject.
+  return new Promise((resolve, reject) => {
+    if (elements.length === 0) {
+      return reject(new Error("No placements found."));
+    }
```

Comment:
Yeah not sure. It seems like something to support, this will find all placements and perform individual requests for each. I'm not sure how frequent that will be, but seems like an okay default experience. I could see it being confusing if we only support a single placement.

Reply:
Currently, we only support a single placement. That's how we have pitched this to our advertisers.

Different ad networks handle this in different ways. The most common that I've seen is to have 2 placements but ensure that both go to the same advertiser so that the ads work together. I think supporting 1 for now and possibly expanding that is a good way forward.

Reply:
I'll follow up with a patch for this, I have some conflicting changes in my next PR.